### ISSUE TITLE
make tre-deploy fails updating SharedSubnet as operation in progress

### ISSUE THREAD

**Issue by marrobi** [microsoft/AzureTRE#1823]
**Describe the bug**
When running `make tre-deploy` occasionally get the below error. I expect this needs a dependancy somewhere?

 ```
│   with azurerm_cosmosdb_account.tre-db-account,
│   on statestore.tf line 1, in resource "azurerm_cosmosdb_account" "tre-db-account":
│    1: resource "azurerm_cosmosdb_account" "tre-db-account" {
│ 
╵
╷
│ Error: updating Network Security Group Association for Subnet: (Name "SharedSubnet" / Virtual Network Name "vnet-mrdeveast2" / Resource Group "rg-mrdeveast2"): network.SubnetsClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=<nil> Code="AnotherOperationInProgress" Message="Another operation on this or dependent resource is in progress. To retrieve status of the operation use uri
```

**Comment by ross-p-smith** [comment]
I've recently come across this - see #2222

**Comment by marrobi** [comment]
Believe to be fixed in #2414 . If reoccurs we can reopen.
